utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/KafkaUniversalComponentTransformer.scala

```diff
@@ -98,15 +117,31 @@ abstract class KafkaUniversalComponentTransformer[T, TN <: TopicName: TopicValid
   protected def getVersionParam(
       preparedTopic: PreparedKafkaTopic[TN],
   )(implicit nodeId: NodeId): WithError[ParameterCreatorWithNoDependency with ParameterExtractor[String]] = {
-    val versions = schemaRegistryClient.getAllVersions(preparedTopic.prepared.toUnspecialized, isKey = false)
-    (versions match {
-      case Valid(versions) => Writer[List[ProcessCompilationError], List[Integer]](Nil, versions)
-      case Invalid(e) =>
-        Writer[List[ProcessCompilationError], List[Integer]](
-          List(CustomNodeError(e.getMessage, Some(topicParamName))),
-          Nil
-        )
-    }).map(getVersionParam)
+    val topicsWithSchema = topicSelectionStrategy.getTopics(schemaRegistryClient)
+    if (topicsWithSchema.exists(_.contains(preparedTopic.prepared.topicName.toUnspecialized))) {
+      val versions = schemaRegistryClient.getAllVersions(preparedTopic.prepared.toUnspecialized, isKey = false)
+      (versions match {
+        case Valid(versions) => Writer[List[ProcessCompilationError], List[Integer]](Nil, versions)
+        case Invalid(e) =>
+          Writer[List[ProcessCompilationError], List[Integer]](
+            List(CustomNodeError(e.getMessage, Some(topicParamName))),
+            Nil
+          )
+      }).map(getVersionParam)
+    } else {
+      val versionValues = List(
+        FixedExpressionValue("'Json'", "Json"),
```

Comment:
IMO we should not use the same `version` parameter for selecting content type for schemaless topics. Based on existence of schema in schema registry we should render `version` or `contentType` parameter in the source.